CenterForOpenScience/osf.io
# [Feature] metatags for zotero [OSF-8709]

## Purpose

Allow Zotero to pick up more metadata about osf projects 

## Changes

Add meta tags for `authors`, `institutions`, `date modified`, `tags`, `relations`, `identifiers`
using combo of Google Highwire Press, Dublin Core, and Open Graph tags.
Both Highwire Press and Dublin Core tags are used by Zotero.
Facebook, Twitter, linkedin use Open Graph Tags.

```mako
    <meta name="citation_title" content="${self.title_meta()}" />
    %if self.identifier_meta():
        <meta name="citation_doi" content="${self.identifier_meta()['doi']}" />
        <meta name="dc.identifier" content="${self.identifier_meta()['doi']}" />
        <meta name="dc.identifier" content="${self.identifier_meta()['ark']}" />
    %endif
    <meta name="citation_publisher" content="Open Science Framework" />
    %for institution in self.institutions_meta()[:10]:
        <meta name="citation_author_institution" content="${institution}" />
    %endfor
    %for rel in self.relations_meta():
        %if rel:
            <meta name="dc_relation" scheme="DCTERMS.URI" content="${rel}" />
        %endif
    %endfor
    <meta name="dc.abstract" content="${self.description_meta()}" />
    <meta name="dc.license" content="${self.license_meta()}" />
    <meta name="dc.date" content="${self.date_meta()}" />
    <meta name="dc.publisher" content="Open Science Framework" />
    <meta name="dc.language" content="en" />


    
    <meta property="og:ttl" content="3" />
    <meta property="og:site_name" content="Open Science Framework" />
    <meta property="og:url" content="${self.url_meta()}" />
    <meta property="og:title" content="${self.title_meta()}" />
    <meta property="og:description" content="${self.description_meta()}" />
    <meta property="og:image" content="${self.image_meta()}" />
    <meta property="og:image:type" content="image/png" />
    <meta property="og:image:width" content="363" />
    <meta property="og:image:height" content="242" />
    <meta property="og:image:alt" content="Open Science Framework" />
    <meta property="article:modified_time" content="${self.date_meta()}" />
    <meta property="article:section" content="${self.category_meta()}" />
    <meta property="og:type" content="article"/>
    %for author in self.authors_meta()[:10]:
        <meta name="dc.creator" content="${author}" />
        <meta name="citation_author" content="${author}" />
        <meta property="article:author" content="${author}" />
    %endfor
    %for tag in self.keywords_meta()[:10]:
        <meta name="citation_keywords" content="${tag}" />
        <meta property="article:tag" content="${tag}" />
    %endfor
    
    <meta name="twitter:card" content="summary">
    <meta name="twitter:site" content="@OSFramework">
    <meta name="twitter:creator" content="@OSFramework">
```
## Ticket

[OSF-8709](https://openscience.atlassian.net/browse/OSF-8709)

## Before
![before-metatags](https://user-images.githubusercontent.com/4511563/31721855-7d8515ee-b3e8-11e7-8f18-dc3455325ac7.gif)

## After
![after-metatags](https://user-images.githubusercontent.com/4511563/31721874-8337c112-b3e8-11e7-82b8-ce0fdcfef3c9.gif)

## QA Considerations
Currently, unable to capture tags, relations with Zotero plugin. I submit a ticket to Zotero ppl, see [here](https://forums.zotero.org/discussion/comment/288651)
